app/models/audit/event/can_fail.rb

```diff
@@ -0,0 +1,49 @@
+require 'English'
+
+module Audit
+  class Event
+    module CanFail
+      def message
+        if success
+          success_message
+        else
+          [failure_message, error_message].compact.join(': ')
+        end
+      end
+      
+      def structured_data 
+        super.deep_merge SDID::ACTION => { result: success_text }
+      end
+      
+      def severity
+        success ? Syslog::LOG_INFO : Syslog::LOG_WARNING
+      end
+
+      def self.included other
+        other.class_eval do
+          field success: true, error_message: nil
+          abstract_field :success_message, :failure_message
+          extend CanFail::ClassMethods
+        end
+      end
+
+      module ClassMethods
+        # Automatically initialize an instance with data from currently handled
+        # exception (if any).
+        def with_exception **parameters
```

Comment:
I think invocations of this method would read better if it was called `new_with_exception`.

Reply:
Thanks, fixed.